lib/kube/proxy/sess.go

```diff
@@ -364,6 +380,22 @@ func newSession(ctx authContext, forwarder *Forwarder, req *http.Request, params
 	return s, nil
 }
 
+// disconnectPartyOnErr is called when any party connection returns an error.
+// It is used to properly handle client disconnections.
+func (s *session) disconnectPartyOnErr(idString string, err error) {
+	if idString == sessionRecorderID {
+		s.log.Error("Failed to write to session recorder, closing session.")
+		s.Close()
+	}
+
+	s.log.Errorf("Encountered error: %v with party %v. Disconnecting them from the session.", err, idString)
+	id, _ := uuid.Parse(idString)
+
+	if err = s.leave(id); err != nil {
+		s.log.Errorf("Failed to disconnect party %v from the session: %v.", idString, err)
```

Comment:
If `Parse` fails, `id` will be an empty string, and this log message will be pretty confusing.